### PR TITLE
Skip mutated struct properties in bpdoc.

### PR DESCRIPTION
**Pull request by gyias**
This change fixes a bug where the `blueprint:mutated` tag was ignored
when used for struct or interface properties. It also fixes a dormant
ExcludeByTag func bug.

Test: m soong_docs, properties_test.go, bpdoc_test.go
Change-Id: I926de2aa203ec552ced897174e9b78e817701a7d

### CODE REVIEW

File: bootstrap/bpdoc/bpdoc.go [hunk 1]
```diff
@@ -173,10 +173,14 @@ func nestedPropertyStructs(s reflect.Value) map[string]reflect.Value {
 				// The field is not exported so just skip it.
 				continue
 			}
+			if hasTag(field.Tag, "blueprint", "mutated") {
+				continue
+			}
 
 			fieldValue := structValue.Field(i)
+			kind := fieldValue.Kind()
```
Comment (danw) [inline bootstrap/bpdoc/bpdoc.go]: Why this change?

Comment (gyias) [reply bootstrap/bpdoc/bpdoc.go]: Leftover from debugging endeavor. Removed.

File: bootstrap/bpdoc/properties.go [hunk 2]
```diff
@@ -250,17 +250,23 @@ func filterPropsByTag(props *[]Property, key, value string, exclude bool) {
...
-			}
+		if hasTag(x.Tag, key, value) == !exclude {
+			filtered = append(filtered, x)
 		}
 	}
 
 	*props = filtered
 }
 
+func hasTag(tag reflect.StructTag, key, value string) bool {
```
Comment (colincross) [inline bootstrap/bpdoc/properties.go]: Can this reuse proptools.HasTag?

Comment (gyias) [reply bootstrap/bpdoc/properties.go]: Not entirely since proptools.HasTag accepts reflect.StructField, and the above call fetches the tag from bpdoc.Property. I replaced the one in bpdoc.go, though.

**Comment by gyias** [comment]
> Can you squash the changes together?
